### ISSUE TITLE
Build imgui-bundle example successfully into excutable, but crashed at getting windows WndProc hook

### ISSUE THREAD

**Issue by hugle** [Nuitka/Nuitka#2812]
## Nuitka Version

```
2.1.5
Commercial: None
Python: 3.11.8 | packaged by conda-forge | (main, Feb 16 2024, 20:40:50) [MSC v.1937 64 bit (AMD64)]
Flavor: Anaconda Python
Executable: C:\Users\ABC\AppData\Local\Programs\miniconda3\envs\test\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10
Version C compiler: C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.39.33519\bin\Hostx64\x64\cl.exe (cl 14.3).
```

I am able to run the example [`imgui_example_glfw_opengl3.py`](https://github.com/pthom/imgui_bundle/blob/main/bindings/imgui_bundle/demos_python/demos_immapp/imgui_example_glfw_opengl3.py) from `imgui_bundle` package without issue by executing:

```
python imgui_example_glfw_opengl3.py
```
But when I was trying to build the example `imgui_example_glfw_opengl3.py` into an executable app using Nuitka. Below are my conda env packages:

```
Package                   Version
------------------------- -----------
Nuitka                    2.1.5
PyOpenGL                  3.1.7
glfw                      2.7.0
imgui-bundle              1.3.0
...
(and other packages come after running: pip install imgui-bundle)
```
With commands: 
```
nuitka imgui_example_glfw_opengl3.py --standalone --follow-imports --python-flag=-O --python-flag=-v --output-dir=dist
```
Nuitka produced an executable successfully, but when I run it, it stopped at following debug message:

```
(... lots of message ...)
import glfw_preview # denied responsibility
import cffi # considering responsibility (find_spec)
import cffi # denied responsibility
Loaded glfw
import 'glfw' # <class 'nuitka_module_loader'>
Loaded imgui_bundle.glfw_utils
import 'imgui_bundle.glfw_utils' # <class 'nuitka_module_loader'>
Loaded imgui_bundle
import 'imgui_bundle' # <class 'nuitka_module_loader'>
Traceback (most recent call last):
  File "C:\Users\ABC\source\imgui_sample\dist\imgui_opengl.dist\imgui_example_glfw_opengl3.py", line 289, in <module>
  File "C:\Users\ABC\source\imgui_sample\dist\imgui_opengl.dist\imgui_example_glfw_opengl3.py", line 102, in main
RuntimeError: IM_ASSERT( bd->PrevWndProc != nullptr )   ---   imgui_impl_glfw.cpp:651
# clear builtins._
# clear sys.path
# clear sys.argv
# clear sys.ps1
# clear sys.ps2
(... lots of message ...)
```

Trace to the source, the message comes from the following:

```python
    # You need to transfer the window address to imgui.backends.glfw_init_for_opengl
    # proceed as shown below to get it.
    window_address = ctypes.cast(window, ctypes.c_void_p).value    <---- getting glfw window address
    imgui.backends.glfw_init_for_opengl(window_address, True)      <---- Executable stopped here
```

Which is caused by getting null pointer at line *651* from [`imgui_impl_glfw.cpp`](https://github.com/pthom/imgui/blob/5f0d4186e871e90b0e42ce4d7e19967861c60ded/backends/imgui_impl_glfw.cpp):

```cpp
649 #ifdef _WIN32
650     bd->PrevWndProc = (WNDPROC)::GetWindowLongPtrW((HWND)main_viewport->PlatformHandleRaw, GWLP_WNDPROC);
651     IM_ASSERT(bd->PrevWndProc != nullptr);    <--------------------- Null pointer here
652     ::SetWindowLongPtrW((HWND)main_viewport->PlatformHandleRaw, GWLP_WNDPROC, (LONG_PTR)ImGui_ImplGlfw_WndProc);
#endif
```

In summary, my issue is that directly running python script has no issue. The executable is generated successfully by Nuitka without error. But crashed due to null WndProc at the source described above.
My guess is could it be the way Nuitka handles `ctypes.c_void_p` caused the issue?

Thanks for the help!

**Comment by kayhayen** [comment]
This could well be some form of resource making it not use what it expects to be used. I do not believe in ctypes usage being to blame, that's known to be all flawless. But for acceleration and things, executables sometimes need to set something. I will look at the resource of Anaconda Python and Nuitka there. It has become rare, and glfw is known to work. But I recall it has a plugin, but I forget now if it's maybe on by default, or if you didn't get a warning, or it wasn't detected. That would be the other angle.

**Comment by hugle** [comment]
Thanks @kayhayen , let me know if you need any information for troubleshooting.

**Comment by hugle** [comment]
I have tried again. There is no issue building following simple glfw example using Nuitka, so might not be an issue from glfw. For some reason, glfw window reference is going.

```python
if __name__ == '__main__':
    import sys
    import glfw
    import OpenGL.GL as gl

    def on_key(window, key, scancode, action, mods):
        if key == glfw.KEY_ESCAPE and action == glfw.PRESS:
            glfw.set_window_should_close(window,1)

    # Initialize the library
    if not glfw.init():
        sys.exit()

    # Create a windowed mode window and its OpenGL context
    window = glfw.create_window(640, 480, "Hello World", None, None)
    if not window:
        glfw.terminate()
        sys.exit()

    # Make the window's context current
    glfw.make_context_current(window)

    # Install a key handler
    glfw.set_key_callback(window, on_key)

    # Loop until the user closes the window
    while not glfw.window_should_close(window):
        # Render here
        width, height = glfw.get_framebuffer_size(window)
        ratio = width / float(height)
        gl.glViewport(0, 0, width, height)
        gl.glClear(gl.GL_COLOR_BUFFER_BIT)
        gl.glMatrixMode(gl.GL_PROJECTION)
        gl.glLoadIdentity()
        gl.glOrtho(-ratio, ratio, -1, 1, 1, -1)
        gl.glMatrixMode(gl.GL_MODELVIEW)
        gl.glLoadIdentity()
        # gl.glRotatef(glfw.glfwGetTime() * 50, 0, 0, 1)
        gl.glBegin(gl.GL_TRIANGLES)
        gl.glColor3f(1, 0, 0)
        gl.glVertex3f(-0.6, -0.4, 0)
        gl.glColor3f(0, 1, 0)
        gl.glVertex3f(0.6, -0.4, 0)
        gl.glColor3f(0, 0, 1)
        gl.glVertex3f(0, 0.6, 0)
        gl.glEnd()

        # Swap front and back buffers
        glfw.swap_buffers(window)

        # Poll for and process events
        glfw.poll_events()

    glfw.terminate()
```

**Comment by hugle** [comment]
Because the following is a python binding call of `glfw_init_for_opengl` :

```python
    # You need to transfer the window address to imgui.backends.glfw_init_for_opengl
    # proceed as shown below to get it.
    window_address = ctypes.cast(window, ctypes.c_void_p).value    <---- getting glfw window address
    imgui.backends.glfw_init_for_opengl(window_address, True)      <---- Executable stopped here
```

The function is produced from `pybind11` at [pybind_igui_backends.cpp](https://github.com/pthom/imgui_bundle/blob/85d3fafa16794f0eb3d8c6fcf1a7918550ee9591/external/imgui/bindings/pybind_imgui_backends.cpp#L82) :

```cpp
    //
    // <bindings for imgui_impl_glfw.h
    //
    auto glfw_init_for_opengl = [](size_t window_address, bool install_callbacks) {
        return ImGui_ImplGlfw_InitForOpenGL((GLFWwindow*)window_address, install_callbacks);
    };
    m.def("glfw_init_for_opengl", glfw_init_for_opengl, py::arg("window_address"), py::arg("install_callbacks"));     // <----- binding issue?
    m.def("glfw_init_for_open_gl", glfw_init_for_opengl, py::arg("window_address"), py::arg("install_callbacks")); // legacy synonym

```

Could it be binding issue?

**Comment by kayhayen** [comment]
Check with --list-package-dlls=imgui-bundle or whatever the package name of that PyPI package is, and if there are DLLs, they are indeed likely to cause this. Also maybe the dependency on OpenGL is somewhat hidden from the plugin, and a package needs to trigger it as well.

If you are available to do a PR, this can be fixed by adding to the yaml configuration, described here: [https://nuitka.net/doc/nuitka-package-config.html](https://nuitka.net/doc/nuitka-package-config.html)

And here is more of a walk through style post: [https://nuitka.net/posts/nuitka-package-config-kickoff.html](https://nuitka.net/posts/nuitka-package-config-kickoff.html)

Here are a example PRs for other packages:

[#2030](https://github.com/Nuitka/Nuitka/pull/2030) 
[#1993](https://github.com/Nuitka/Nuitka/pull/1993)

**Comment by hugle** [comment]
Nuitka has done things right, it has included both the DLL files from `glfw` python package and the version of `glfw3.dll` from `imgui-bundle`. So it was not that dll file was missing from building into *exe* using Nuitka.

**Comment by kayhayen** [comment]
So, this is directly using glfw, so the plugin ought to trigger indeed, this is all pretty standard actually.

**Comment by kayhayen** [comment]
And on my Linux indeed it works, with 2.6.2 glfw, updating to 2.7 now. Maybe they changed something on their OS plugins side again and that's what we are dealing with, but 2.7.0 is fine on Linux too.

**Comment by kayhayen** [comment]
This is lacking an example, I don't want to guess how correct or incorrect you packages the example, that is full of things that need to be adapted, e.g. font data files. Without a reproducer, I cannot work on this.

At the end of the day, the difference may come down to this

```
python3.10 bin/nuitka --list-package-dlls=imgui_bundle
Nuitka-Tools: Checking package directory 'C:\Python310_64\lib\site-packages\imgui_bundle' ..
C:\Python310_64\lib\site-packages\imgui_bundle
  glfw3.dll
  libglfw.3.dylib
  libglfw.3.so
  opencv_world460.dll
Nuitka-Tools: Found 4 DLLs.

kayha@Neyah MINGW64 ~/repos/Py2C (main)
$ python3.10 bin/nuitka --list-package-dlls=glfw
Nuitka-Tools: Checking package directory 'C:\Python310_64\lib\site-packages\glfw' ..
C:\Python310_64\lib\site-packages\glfw
  glfw3.dll
  msvcr110.dll
Nuitka-Tools: Found 2 DLLs.
```

These DLLs might be different, and imgui_bundle might try and get its on one loaded (ugly how they contain all DLLs) or maybe it's just the opencv DLL that is missing, but without a reproducer, that's not happening.

**Comment by hugle** [comment]
Thanks for working on this, I modify the font paths below by taking the same the runnable script:

[demo.zip](https://github.com/Nuitka/Nuitka/files/15072340/demo.zip)

**Comment by kayhayen** [comment]
I successfully reproduced it using the demo, when trying to google the code, it doesn't exist anymore, so it's a bit hard to know what exactly it thinks went wrong, and maybe if this would be different if that new code of `imgui_bundle` was used. I also tried playing around with not using two DLLs for glfw using either one of them, but it didn't change stuff, also the opencv DLL had (of course?) no impact either.

The DLL load tracing will have to help us here, experiments were unsuccessful. It could totally be that also mere strace will help us there too.

**Comment by typenoob** [comment]
Using glfw together with imgui-bundle requires both to load the same GLFW library. Nuitka should honor imgui-bundle's PYGLFW_LIBRARY and ship its vendored GLFW instead of forcing the glfw package DLL.
